webapp/src/ts/modules/messages/messages.component.ts

```diff
@@ -48,6 +53,11 @@ export class MessagesComponent implements OnInit, OnDestroy {
   ngOnInit(): void {
     this.subscribeToStore();
     this.tourService.startIfNeeded(this.route.snapshot);
+    if (!this.sessionService.isOnlineOnly()) {
+      this
+        .getCurrentLineageLevel()
+        .then(currentLevel => this.currentLevel = currentLevel);
```

Comment:
@latin-panda you have way more front-end experience then I do, so you might know if this is worth worrying about!  I got to thinking about how loading the `currentLevel` here is async and so technically there is nothing stopping the `updateConversations` code below running before we actually get the `currentLevel` set.  

Would it be better to just assign `currentLevel` to a promise that we have to wait for down in `updateConversations`?  e.g this code would become something like `this.currentLevel = this.getCurrentLineageLevel();`  Then down in  `updateConversations` we do something like `Promise.all([this.messageContactService.getList(), this.currentLevel]).then(....`

Reply:
Yes thank you, that's possible, I was considering something similar yesterday, but your suggestion about assigning `currentLevel` to a promise is simpler and what we usually do. 

@elvisdorkenoo is it making sense?

Reply:
yeah totally makes sense, thanks.